microsetta_interface/templates/new_results_page.jinja2

```diff
@@ -91,24 +91,164 @@
+        // n_bacteria_background, n_archaea_background,
+        // age_nearest_neighbor, more_or_less_sweets_nearest_neighbor
+
+        function retrieveText(method, url, data){
+            return $.ajax({
+                method: method,
+                url: url,
+                data: data
+            }).fail(function(result, textStatus, errorThrown){
+                console.log("Couldn't retrieve text for " + method + " " + url)
+            })
+        }
+
+        function setInfoText(resultSelector){
+            return function(result, textStatus, jqXHR){
+                return $(resultSelector).text(result).removeClass("spinner-grow spinner-grow-sm")
+            }
+        }
+
+        function retrieveNeighbors(state, k){
+            url = state.public_endpoint + '/dataset/' + state.dataset + '/diversity/beta/' + state.beta_metric + '/nearest'
+            return $.ajax({
+                method: "GET",
+                url: url,
+                data: {
+                    sample_id: state.barcode_prefix + state.sample_id,
+                    k:k
+                }
+            }).fail(function(result, textStatus, errorThrown){
+                console.log("Couldn't retrieve neighbors" + textStatus)
+            })
+        }
+
+        function queryNeighborAndSelfMetadata(state, metadata_cats){
+            return function(result, textStatus, jqXHR)
+            {
+                var toQuery = result;
+                toQuery.push(state.barcode_prefix + state.sample_id);
+
+                return $.ajax(
+                {
+                    method: "POST",
+                    url: state.public_endpoint + "/metadata/values?" + $.param({ cat: metadata_cats }, true), // The true indicates traditional mode, which produces cat=blah&cat=blah rather than cat[]=blah&cat[]=blah which crashes the server.
+                    data: JSON.stringify(toQuery),
+                    contentType: "application/json"
+                }).fail(function(result, textStatus, errorThrown){
+                    console.log("Couldn't retrieve metadata for neighbors and self " + errorThrown)
+                }).then(function(result, textStatus, jqXHR){
+                    thenner = {}
+                    thenner.then = function(func){
+                        // Need to pass on the sample ids. Argh.
+                        return func(toQuery, result);
+                    }
+                    return thenner;
+                });
+            };
+        }
+
+        function fillResults(state, selectors)
+        {
+            return function(samplesQueried, sampleResults)
+            {
+                // First selector, age, Second selector, sweets
+                // No similarity between the two, so we just have to handle both
+                me = samplesQueried.pop()
+                my_data = sampleResults.pop()
+
+                if (me != state.barcode_prefix + state.sample_id)
```

Comment:
Not obvious if this is `(me != state.barcode_prefix) + state.sample_id` or `me != (state.barcode_prefix + state.sample_id)`. Javascript being crazy, I think it would be better to be defensive